src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Unix.cs

```diff
@@ -267,6 +267,20 @@ protected override void Dispose(bool disposing)
                         // name will be removed immediately.
                         Interop.Sys.Unlink(_path); // ignore errors; it's valid that the path may no longer exist
                     }
+
+                    // Closing the file handle can fail, e.g. due to out of disk space
+                    // Throw these errors as exceptions when disposing
+                    if (_fileHandle != null && !_fileHandle.IsClosed && disposing)
+                    {
+                        SafeFileHandle.t_lastCloseErrorInfo = null;
+
+                        _fileHandle.Dispose();
+
+                        if (SafeFileHandle.t_lastCloseErrorInfo != null)
+                        {
+                            throw Interop.GetExceptionForIoErrno(SafeFileHandle.t_lastCloseErrorInfo.GetValueOrDefault(), _path, isDirectory: false);
```

Comment:
Since the `t_lastCloseErrorInfo` field is static and ThreadStatic, it needs to be accessed via the class name, not the handle instance, so I corrected that.